config/arch.aarch64

```diff
@@ -13,7 +13,7 @@
     generic|cortex-a35|cortex-a53|cortex-a57|cortex-a72|exynos-m1|qdf24xx|thunderx|xgene1|cortex-a57.cortex-a53|cortex-a72.cortex-a53)
       TARGET_SUBARCH=aarch64
       TARGET_VARIANT=armv8-a
-      TARGET_ABI=eabi
+      TARGET_ABI=eabihf
```

Comment:
huh? there is no soft float abi on aarch64.

Reply:
Sooo... are you agreeing with this change? Regardless of whether there is a soft float abi or not, the suggestion is that we should be including the `hf` suffix in the hard float abi.

Reply:
The aarch64 change is probably unnecessary. zlib-ng's hardfloat test appears to be looking for eabihf only for arm. There's a check further down which assumes floating point is true on aarch64. No floating point support (march=armv8-a+nofp(+nosimd?)) is possible, but I haven't seen it in the wild.

Reply:
It looks required to me - this is the zlib-ng code in question:

https://github.com/zlib-ng/zlib-ng/blob/3ea747396dcc3d63924eedbdf05bacf7866e1b29/CMakeLists.txt#L202-L214

where `BASEARCH_ARM_FOUND` is:

https://github.com/zlib-ng/zlib-ng/blob/3ea747396dcc3d63924eedbdf05bacf7866e1b29/cmake/archdetect.cmake#L51-L53

Therefore setting hard or soft float by matching `eabihf` occurs for BOTH `arm` AND ` aarch64`.

I can test this later today and confirm what happens (I may have already done this, but forgotten!) - I can build Allwinner/A64 (which is using aarch64) with `eabi` (ie. without this PR) and I'm fairly certain zlib-ng will try to use soft float. Allwinner/A64 definitely builds successfully with this PR and zlib-ng.

Reply:
By the way I'm not saying what zlib-ng is doing is entirely correct or sensible (it's certainly wrong in places with bias towards armv8 etc.), but if hard float should be reflected in the os triplet then that's something we're not currently doing and can add easily enough hopefully without any issues.

Reply:
arm change is ok. aarch64 is not and should not be added. it makes no sense to add just because random library's cmake buildsystem is (semi-)broken or makes wrong assumptions.

Reply:
Yes, we've been discussing this on LE Slack and have come to the same conclusion, I'll remove the aarch64 change. BTW I've posted our/my concerns to zlib-ng, [here](https://github.com/zlib-ng/zlib-ng/issues/478#issuecomment-559167356) - would appreciate your thoughts if want to post there! :smile:

Reply:
aarch64 change now dropped.

Reply:
well, they could just trust distro's compiler / CFLAGS.

Reply:
Yes, that's what I suggested. :smile: